src/game_api/render_api.cpp

```diff
@@ -137,3 +139,198 @@ const char** RenderAPI::load_texture(std::string file_name)
     auto load_texture = (LoadTextureFunT*)get_load_texture();
     return load_texture(render_api, &file_name, 1);
 }
+
+using VanillaRenderHudFun = void(void*, float, float);
+VanillaRenderHudFun* g_render_hud_trampoline{nullptr};
+void render_hud(void* hud_data, float y, float opacity)
+{
+    g_render_hud_trampoline(hud_data, y, opacity);
+
+    LuaBackend::for_each_backend(
+        [&](LuaBackend& backend)
+        {
+            backend.vanilla_render();
+            return true;
+        });
+}
+
+void init_render_hud_hook()
+{
+    static size_t func_render_hud = 0;

```

Comment:
Also all this stuff does not need to be static, the function is called only once